codalab/lib/worksheet_util.py

```diff
@@ -481,17 +482,18 @@ def format_metadata(metadata):
                 metadata[name] = apply_func(func, metadata[name])
 
 
-def canonicalize_schema_item(args):
+def canonicalize_schema_item(args, from_schema_name=None):
```

Comment:
Why do we need to return from_schema_name if that's what the user is already passing into this function itself?

Reply:
Because the function returns a tuple which can't be later modified, so we have to inject the from_schema_name information at this time. T